fixups/RegLegacyFixups/RegistryFixups.cpp

```diff
@@ -257,6 +261,28 @@ REGSAM RegFixupSam(std::string keypath, REGSAM samDesired, DWORD RegLocalInstanc
     return samModified;
 }
 
+template <typename CharT>
+const char* ReplaceRegistryQueryPath(PHKEY key, const CharT* subKey)
+{
+    std::string keypath = InterpretKeyPath(*key) + "\\" + InterpretStringA(subKey);
+
+    std::string normalizedKeypath = ReplaceRegistrySyntax(keypath);
+    size_t subkeyOffset = normalizedKeypath.find_first_of('\\');
+    if (subkeyOffset != std::wstring_view::npos)
+    {
+        std::string requestedSubkey = normalizedKeypath.substr(subkeyOffset + 1);
+        auto it = g_regRedirectedHivePaths.find(requestedSubkey);
+        if (it != g_regRedirectedHivePaths.end())
+        {
+            *key = HKEY_CURRENT_USER;
```

Comment:
Why modify key to HKCU?

Reply:
Because HKLM cannot be written to without admin permission. Don't want to force that constraint to application.